Workiva/over_react
# FED-756 Allow analyzer 2.x, fix analyzer plugin in newer SDKs

## Motivation
We want to pull in newer analyzer versions, and unlocking analyzer 2.x is the next step.

Also, while testing I noticed that the analyzer plugin doesn't correctly start in IDEs in Dart 2.18 ☹️. Since this PR touches analyzer_plugin, I figured I'd fix that so we can manually verify changes.

## Changes
- Widen analyzer range to allow 2.x
- Reinstate validate_analyzer CI check to ensure over_react is compatible with both 1.x and 2.x
- Mockito
    - Pin mockito (in dev_dependencies) to work around compilation issue
    - Remove unnecessary List mock, which was causing build failures (more info in 578132d9eaf0f1df8547937f3c5bc25a103e0926)
- Upgrade analyzer plugin to analyzer 2.x. 
    Supporting both isn't necessary since the plugin is its own separate dependency graph, and supporting one is almost definitely easier. I went ahead and picked 2.x as the one to resolve to, since we'd have to upgrade anyways whenever the main package drops support for 1.x.
    - Fix warnings in caused by upgrade (mostly removal of non-null assertions, since analyzer 2.x promoted some APIs to non-nullable)
    - Copy IgnoreInfo into package, since we were previously importing it from analyzer via a src import, and the API changed to take in `ErrorCode` objects instead of Strings, and we couldn't construct ErrorCodes without more src imports
    - Fix low-hanging deprecation warnings
    - Disable ToggleComponentStatefulness, BoilerplateValidatorDiagnostic, and IncorrectDocCommentLocationDiagnostic ☹️. 
        These all import non-null-safe code from over_react's boilerplate parsing, which apparently prevents the plugin from being able to run in 2.18 SDK, and our previous workaround of a language comment in the plugin's `main` no longer works. Until we're able to migrate that code to null safety (which will hopefully be soonish), we'll need to stop importing these to get the plugin working again.

#### Release Notes
Widen analyzer range to allow 2.x in addition to 1.x

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
        - All CI passes , notably the _Dart CI / validate_analyzer (2.18.7, ^2.0.0)_ run
        - Verify analyzer plugin works in playground in Dart 2.18

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:


[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria
